examples/exex/rollup/src/execution.rs

```diff
@@ -0,0 +1,461 @@
+use alloy_consensus::{SidecarCoder, SimpleCoder};
+use alloy_rlp::Decodable as _;
+use eyre::OptionExt;
+use reth::transaction_pool::TransactionPool;
+use reth_interfaces::executor::BlockValidationError;
+use reth_node_api::{ConfigureEvm, ConfigureEvmEnv};
+use reth_node_ethereum::EthEvmConfig;
+use reth_primitives::{
+    constants,
+    eip4844::kzg_to_versioned_hash,
+    keccak256,
+    revm::env::fill_tx_env,
+    revm_primitives::{CfgEnvWithHandlerCfg, EVMError, ExecutionResult, ResultAndState},
+    Address, Block, BlockWithSenders, Bytes, Hardfork, Header, Receipt, TransactionSigned, TxType,
+    B256, U256,
+};
+use reth_revm::{
+    db::{states::bundle_state::BundleRetention, BundleState},
+    DatabaseCommit, StateBuilder,
+};
+use reth_tracing::tracing::debug;
+
+use crate::{db::Database, RollupContract, CHAIN_ID, CHAIN_SPEC};
+
+/// Execute a rollup block and return (block with recovered senders)[BlockWithSenders], (bundle
+/// state)[BundleState] and list of (receipts)[Receipt].
+pub async fn execute_block<Pool: TransactionPool>(
```

Comment:
this is growing quite large, maybe we can extract some of the evm setup or prechecks into another function?

Reply:
split into more functions